apps/settings/js/panels/about/update_check.js

```diff
@@ -150,7 +154,8 @@ define(function(require) {
      */
     _checkForUpdates: function uc__checkForUpdates() {
       if (!navigator.onLine) {
-        alert(this._('no-network-when-update'));
+        navigator.mozL10n.formatValue('no-network-when-update').then(
+          msg => alert(msg));
```

Comment:
absolutely not. Actually, at this moment there is no way for it to take any time because all this code will be fired after we already translated DOM and displayed the UI (which means that we have all resources in memory).

In the future we would like to consider switching to runtime language fallbacking which means that in error scenarios (when for some reason the string cannot be retrieved/resolved in user selected language) we will load fallback locale and use the string from there, this can take ~100-200ms.

But I don't ever expect a scenario when retrieving a string takes longer than than, and in all non-error scenarios it is literally below 5ms (timeout of a Promise).

So, while I like to think of formatValue as "it should work even if it would take 5s", I don't believe it's a realistic worry.